src/entities/User/routes.ts

```diff
@@ -0,0 +1,125 @@
+import { WithAuth, auth } from 'decentraland-gatsby/dist/entities/Auth/middleware'
+import handleAPI from 'decentraland-gatsby/dist/entities/Route/handle'
+import routes from 'decentraland-gatsby/dist/entities/Route/routes'
+import { hashMessage, recoverAddress } from 'ethers/lib/utils'
+import { Request, Response } from 'express'
+import isEthereumAddress from 'validator/lib/isEthereumAddress'
+
+import { DiscourseService } from '../../services/DiscourseService'
+import { ErrorService } from '../../services/ErrorService'
+import isDebugAddress from '../Debug/isDebugAddress'
+
+import { GATSBY_DISCOURSE_CONNECT_THREAD, MESSAGE_TIMEOUT_TIME } from './constants'
+import UserModel from './model'
+import { ValidationMessage } from './types'
+
+export default routes((route) => {
+  const withAuth = auth()
+  route.get('/user/:address/is-validated', handleAPI(isValidated))
+  route.get('/user/validate', withAuth, handleAPI(getValidationMessage))
+  route.post('/user/validate', withAuth, handleAPI(checkValidationMessage))
+  route.delete('/user/validate', withAuth, handleAPI(deleteValidation))
+})
+
+const VALIDATIONS_IN_PROGRESS: Record<string, ValidationMessage> = {}
+
+function clearValidationInProgress(user: string) {
+  const validation = VALIDATIONS_IN_PROGRESS[user]
+  if (validation) {
+    clearTimeout(validation.messageTimeout)
+    delete VALIDATIONS_IN_PROGRESS[user]
+  }
+}
+
+function formatValidationMessage(address: string, timestamp: string) {
+  return `By signing and posting this message I'm linking my Decentraland DAO account ${address} with this Discourse forum account\n\nDate: ${timestamp}`
+}
+
+async function getValidationMessage(req: WithAuth) {
+  const address = req.auth!
```

Comment:
We should check if there is an address here.

Reply:
It is an endpoint with authentication. If you try to use it without being authenticated, the middleware rejects in advance

![image](https://github.com/decentraland/governance/assets/45410089/a3e35ef6-3031-4d19-881c-7fca48e5961a)

Reply:
And it is the same as we have it everywhere, here is an example:

![image](https://github.com/decentraland/governance/assets/45410089/f7b7d2d6-ce8c-4e8a-b99b-8262585fd3b6)